src/data.ts

```diff
@@ -0,0 +1,126 @@
+import {User, SimpleUser, IdentifiedUser, buildUser, changeUser} from './user';
+import {MapPersistor} from './mappersistor';
+import {Persistor} from './persistor';
+
+/**
+ * Manages a map that holds Users
+*/
+export class UserRepository {
+
+  private persist: Persistor<User>;
+  private users = new Map<number, User >();
+  private idCounter = 0;
+
+  /** 
+   * Creates an instance of UserRepository
+   */
+  public constructor() {
+    //create a new persistor object
+    this.persist = new MapPersistor<User>('./userdata.txt');
+    
+    //have it load the user-map or create a new one, if it doesn't exist
+    this.persist.loadData().then(loadedData => {
+      this.users = loadedData.theMap;
+      this.idCounter = loadedData.idCounter;
+
+      console.log('Data: Initialized data storage');
+      console.log('Data: Currently holding %s user(s)', this.users.size);
+    
+      //Print data for debug purposes
+      this.logAllData();
+    });
+    
+  };
+
+  /** 
+   * Creates a new user and adds him to the map
+   * 
+   * @param {SimpleUser} info An object holding the relevant information for the creation of the User-object
+   */
+  public addUser(info: SimpleUser): void {
+    //create a user with these exact specifications and a current timestamp, also add him to our map
+    this.users.set(this.idCounter, buildUser(info));
```

Comment:
Besides, it is a better style to throw an exception if incoming data do not match your expected structure.